File: python/minimum-bit-flips-to-convert-number.py
class Solution:
    def minBitFlips(self, start: int, goal: int) -> int:
        flips = start ^ goal
        flip_count = 0
        while flips:
            if flips & 1:
                flip_count += 1
            flips = flips >> 1
        return flip_count

File: rust/minimum-bit-flips-to-convert-number.rs
impl Solution {
    pub fn min_bit_flips(start: i32, goal: i32) -> i32 {
        let mut flips = start ^ goal;
        let mut flip_count = 0;
        while 0 < flips {
            if flips & 1 == 1 {
                flip_count += 1;
            }
            flips = flips >> 1;
        }
        return flip_count;
    }
}
